gradle-baseline-java/src/main/java/com/palantir/baseline/plugins/BaselineModuleJvmArgs.java

```diff
@@ -80,37 +94,21 @@ public void apply(Project project) {
             // https://github.com/gradle/gradle/issues/18824
             // However, we set sourceCompatibility in BaselineJavaVersion to opt out of the '--release' flag.
             project.getExtensions().getByType(SourceSetContainer.class).configureEach(sourceSet -> {
+                Provider<String> annotationProcessorConfigurationName =
+                        project.provider(sourceSet::getAnnotationProcessorConfigurationName);
+
                 TaskProvider<JavaCompile> javaCompileProvider =
                         project.getTasks().named(sourceSet.getCompileJavaTaskName(), JavaCompile.class);
                 javaCompileProvider.configure(javaCompile -> {
-                    javaCompile
-                            .getOptions()
-                            .getCompilerArgumentProviders()
-                            // Use an anonymous class because tasks with lambda inputs cannot be cached
-                            .add(new CommandLineArgumentProvider() {
-                                @Override
-                                public Iterable<String> asArguments() {
-                                    // The '--release' flag is set when BaselineJavaVersion is not used.
-                                    if (!project.getPlugins().hasPlugin(BaselineJavaVersion.class)) {
-                                        project.getLogger()
-                                                .debug(
-                                                        "BaselineModuleJvmArgs not applying args to compilation task "
-                                                                + "{} on {} due to lack of BaselineJavaVersion",
-                                                        javaCompile.getName(),
-                                                        project);
-                                        return ImmutableList.of();
-                                    }
-                                    ImmutableList<String> arguments =
-                                            collectCompilationArgs(project, extension, sourceSet);
-                                    project.getLogger()
-                                            .debug(
-                                                    "BaselineModuleJvmArgs compiling {} on {} with exports: {}",
-                                                    javaCompile.getName(),
-                                                    project,
-                                                    arguments);
-                                    return arguments;
-                                }
-                            });
+                    ModuleJvmArgsArgumentProvider provider = newModuleJvmArgsArgumentProvider(
+                            project,
+                            extension,
+                            getConfigurations().getByName(annotationProcessorConfigurationName.get()));
```

Comment:
Ideally don't call `getByName` at configuration time, as it realises the Configuration (they can now be lazy). Instead called `named` and pass a provider, which you can then give to `newModuleJvmArgsArgumentProvider` (which might want to take an `Object` like `.from` does as other places give it a raw `FileCollection`).